.github/CONTRIBUTOR_GUIDE/CONTRIBUTING.md

```diff
@@ -0,0 +1,108 @@
+
+# Community Scripts Contribution Guide
+
+## **Welcome to the communty-scripts Repository!** 
+📜 These documents outline the essential coding standards for all our scripts and JSON files. Adhering to these standards ensures that our codebase remains consistent, readable, and maintainable. By following these guidelines, we can improve collaboration, reduce errors, and enhance the overall quality of our project.
+
+### Why Coding Standards Matter
+
+Coding standards are crucial for several reasons:
+
+1. **Consistency**: Consistent code is easier to read, understand, and maintain. It helps new team members quickly get up to speed and reduces the learning curve.
+2. **Readability**: Clear and well-structured code is easier to debug and extend. It allows developers to quickly identify and fix issues.
+3. **Maintainability**: Code that follows a standard structure is easier to refactor and update. It ensures that changes can be made with minimal risk of introducing new bugs.
+4. **Collaboration**: When everyone follows the same standards, it becomes easier to collaborate on code. It reduces friction and misunderstandings during code reviews and merges.
+
+### Scope of These Documents
+
+These documents cover the coding standards for the following types of files in our project:
+
+- **`install/$AppName-install.sh` Scripts**: These scripts are responsible for the installation of applications.
+- **`ct/$AppName.sh` Scripts**: These scripts handle the creation and updating of containers.
+- **`json/$AppName.json`**: These files store structured data and are used for the website.
+
+Each section provides detailed guidelines on various aspects of coding, including shebang usage, comments, variable naming, function naming, indentation, error handling, command substitution, quoting, script structure, and logging. Additionally, examples are provided to illustrate the application of these standards.
+
+By following the coding standards outlined in this document, we ensure that our scripts and JSON files are of high quality, making our project more robust and easier to manage. Please refer to this guide whenever you create or update scripts and JSON files to maintain a high standard of code quality across the project. 📚🔍
+
+Let's work together to keep our codebase clean, efficient, and maintainable! 💪🚀
+
+
+## Getting Started
+
+Before contributing, please ensure that you have the following setup:
+
+1. **Visual Studio Code** (recommended for script development)
+2. **Recommended VS Code Extensions:**
+   - [Shell Syntax](https://marketplace.visualstudio.com/items?itemName=bmalehorn.shell-syntax)
+   - [ShellCheck](https://marketplace.visualstudio.com/items?itemName=timonwong.shellcheck)
+   - [Shell Format](https://marketplace.visualstudio.com/items?itemName=foxundermoon.shell-format)
```

Comment:
I like the recommendation overall. However, i think many engineers have their opinionated IDE. Thats why I think we should not "phrase it as mandatory" (this is how I read it).

What I can think about: Having a section like:

```
## IDE setup

Depending on your IDE, here are some recommendations for IDE configurations supporting you in development.

### VSCode

<Recommended extensions here>

### vim

....
```

This way, it feels optional and other people can add other IDEs.
Furthermore, I suggest that this is an own document like `ide.md`.

Reply:
There has been a proposal by @quantumryuu  for this already, have a look here:  https://github.com/community-scripts/ProxmoxVE/tree/new_script_testing/.github/CONTRIBUTOR_GUIDE/dev-scripts
I will start testing, experementing with this when the hollidays are over. 

I also would love to have something simillar to [this](https://community-scripts.github.io/ProxmoxVE/json-editor) for Appname.sh and AppName-install.sh

Reply:
It is only a recommendation, as stated on line 35. I personaly use neovim mostly and adapted it so it fits. But i think that with one example as recommendet is more then enough.

Reply:
Hahaha I had a similar comment written earlier today but decided against adding it. For a couple of reasons:
 1. It is nice to choose a single IDE in your contributor guide, that way you don't have to answer questions from people when they try something else (thus making it non-optional in the guide)
 2. Any seasoned (and thus stubborn) engineer will still try to use their own IDE anyways (looking at myself in the mirror), but they probably also know what they're doing
 3. Any code style guideline should be in place using CI pipelines and with clearly defined tools to be able to enforce that if people use other IDEs they can still contribute while conforming to the same standards